src/Middleware/Spa/SpaProxy/src/SpaProxyLaunchManager.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.SpaProxy
+{
+    internal class SpaProxyLaunchManager : IHostedService, IDisposable
+    {
+        private readonly SpaDevelopmentServerOptions _options;
+        private readonly ILogger<SpaProxyLaunchManager> _logger;
+        private readonly HttpClient _httpClient = new(new HttpClientHandler()
+        {
+            // It's ok for us to do this here since this service is only plugged in during development.
+            ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator
```

Comment:
This is a fairly minor point as it would only matter potentially in the future but... it feels kinda dangerous to have this `_httpClient` lying around in this state, as someone might start using it for other purposes in the future where they actually need cert validation to happen as usual.

Since this `HttpClient` instance is only used to probe for the SPA server startup, it might be nice not to have it as a class-level field, and instead new up an instance each time we're going to do a set of calls into `ProbeSpaDevelopmentServerUrl`. The perf cost of doing that would be nothing in practice.